compiler/rustc_mir/src/interpret/step.rs

```diff
@@ -264,10 +268,12 @@ impl<'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
             NullaryOp(mir::NullOp::SizeOf, ty) => {
                 let ty = self.subst_from_current_frame_and_normalize_erasing_regions(ty);
                 let layout = self.layout_of(ty)?;
-                assert!(
-                    !layout.is_unsized(),
-                    "SizeOf nullary MIR operator called for unsized type"
-                );
+                if layout.is_unsized() {
+                    self.tcx.sess.delay_span_bug(
+                        span,
```

Comment:
you can use `self.frame().current_span()` here, which gets you the same span.

Please leave a `FIXME` comment to remind us to look at this when we fix const eval running early

Reply:
> you can use self.frame().current_span() here, which gets you the same span.

Done.

> Please leave a FIXME comment to remind us to look at this when we fix const eval running early

What should the FIXME say? I don't know what the long-term plan here is.

Reply:
You can reference the new test you added and state that this should be turned into a `span_bug` again once that test will pass with that change.

Reply:
I don't know what you mean by "that change" but I just added a FIXME